packages/lesswrong/components/form-components/ImageUpload.tsx

```diff
@@ -71,26 +84,22 @@ const formPreviewSizeByImageType = {
     width: 153,
     height: 80
   },
+  eventImageId: {
+    width: 320,
+    height: 180
+  }
 }
 
-class ImageUpload extends Component<any,any> {
-  constructor(props, context) {
-    super(props, context);
-    const fieldName = props.name;
-    let imageId = "";
-    if (props.document && props.document[fieldName]) {
-      imageId = props.document[fieldName];
-    }
-    this.state = {
-      imageId,
-    }
-    const addValues = context.updateCurrentValues;
-    const addToSuccessForm = context.addToSuccessForm;
-    addValues({[fieldName]: imageId});
-    addToSuccessForm((results) => this.setImageInfo({} ,""));
-  }
+const ImageUpload = ({name, document, updateCurrentValues, clearField, label, classes}: {
```

Comment:
I had to turn this into a functional component, I think in order to use `useDialog`